biojava-core/src/main/java/org/biojava/nbio/core/util/FileDownloadUtils.java

```diff
@@ -154,13 +164,134 @@ public static void downloadFile(URL url, File destination) throws IOException {
 		copy(tempFile, destination);
 
 		// delete the tmp file
 		tempFile.delete();
 
 	}
+	
+	/**
+	 * Creates validation files beside a file to be downloaded.<br>
+	 * Whenever possible, for a <code>file.ext</code> file, it creates 
+	 * <code>file.ext.size</code> and <code>file.hash</code> for in the same 
+	 * folder where <code>file.ext</code> exists.
+	 * If the file connection size could not be deduced from the URL, no size file is created. 
+	 * If <code>hashURL</code> is <code>null</code>, no hash file is created.
+	 * @param url the remote file URL to download
+	 * @param localDestination the local file to download into
+	 * @param hashURL the URL of the hash file to download. Can be <code>null</code>.
+	 * @param hash The Hashing algorithm. Ignored if <code>hashURL</code> is <code>null</code>.
+	 */
+	public static void createValidationFiles(URL url, File localDestination, URL hashURL, Hash hash){
+		try {
+			URLConnection resourceConnection = url.openConnection();
+			createValidationFiles(resourceConnection, localDestination, hashURL, FileDownloadUtils.Hash.UNKNOWN);
+		} catch (IOException e) {
+			logger.warn("could not open connection to resource file due to exception:\n\t", e.getMessage());
+		}
+	}
+	/**
+	 * Creates validation files beside a file to be downloaded.<br>
+	 * Whenever possible, for a <code>file.ext</code> file, it creates 
+	 * <code>file.ext.size</code> and <code>file.hash_XXXX</code> in the same 
+	 * folder where <code>file.ext</code> exists (XXXX may be DM5, SHA1, or SHA256).
+	 * If the file connection size could not be deduced from the resourceUrlConnection 
+	 * {@link URLConnection}, no size file is created. 
+	 * If <code>hashURL</code> is <code>null</code>, no hash file is created.<br>
+	 * <b>N.B.</b> None of the hashing algorithms is implemented (yet), because we did not need any of them yet.
+	 * @param resourceUrlConnection the remote file URLConnection to download
+	 * @param localDestination the local file to download into
+	 * @param hashURL the URL of the hash file to download. Can be <code>null</code>.
+	 * @param hash The Hashing algorithm. Ignored if <code>hashURL</code> is <code>null</code>.
+	 * @since 6.1.1
+	 */
+	public static void createValidationFiles(URLConnection resourceUrlConnection, File localDestination, URL hashURL, Hash hash){
+		long size = resourceUrlConnection.getContentLengthLong();
+		if(size == -1) {
+			logger.warn("could not find expected file size for resource {}.", resourceUrlConnection.getURL());
+		} else {
+			logger.debug("Content-Length: " + size);
+			File sizeFile = new File(localDestination.getParentFile(), localDestination.getName() + SIZE_EXT);
+			try (PrintStream sizePrintStream = new PrintStream(sizeFile)) {
+				sizePrintStream.print(size);
+				sizePrintStream.close();
+			} catch (FileNotFoundException e) {
+				logger.warn("could not write size validation file due to exception:\n\t", e.getMessage());
```

Comment:
The placeholder `{}` is missing. Also in terms of style I prefer a single line log message (no `\n\t`)

Reply:
Done